pkg/sippyserver/metrics/metrics.go

```diff
@@ -196,7 +196,7 @@ func refreshComponentReadinessMetrics(client *bqclient.Client) error {
 	sampleRelease := apitype.ComponentReportRequestReleaseOptions{
 		Release: next,
 		Start:   today.AddDate(0, 0, -7),
```

Comment:
@stbenjam should this now be `-6` instead of `-7`.

Reply:
I think I can confirm these in debug mode

Reply:
this is what I see in debug:

```
Start : Mon, 02 Oct 2023 00:00:00 +0000
End : Mon, 30 Oct 2023 23:59:59 +0000
Start : Mon, 13 Nov 2023 00:00:00 +0000
End : Mon, 20 Nov 2023 23:59:59 +0000
```

which matches the UI when before you click "Generate Report" and matches the `RequestURL`, in the Network section of the devtools in the browser which shows:

```
baseStartTime=2023-10-02T00:00:00Z
baseEndTime=2023-10-30T23:59:59Z

sampleStartTime=2023-11-13T00:00:00Z
sampleEndTime=2023-11-20T23:59:59Z
```

Reply:
> Start : Mon, 02 Oct 2023 00:00:00 +0000
> End : Mon, 30 Oct 2023 23:59:59 +0000
 
This is 29 days (well, 28d23h59m59s).  4.15 GA'd on 10/31, so I'd expect this to be

Mon 02 Oct 2023 00:00:00 +0000
Tue 31 Oct 2023 23:59:59 +0000

> Start : Mon, 13 Nov 2023 00:00:00 +0000
> End : Mon, 20 Nov 2023 23:59:59 +0000

This is 8 days (7d23h59m59s), I'd expect it to be this instead:

Start : Mon, 14 Nov 2023 00:00:00 +0000
End : Mon, 20 Nov 2023 23:59:59 +0000

Reply:
latest changes now match those values.  I believe the UI still matches though the dates are in local TZ (EST) so the end dau is a day too early on the UI in EST time zone.

Reply:
wait, that first one isn't matching

Reply:
now it matches:

```
Start : Wed, 04 Oct 2023 00:00:00 +0000
End : Tue, 31 Oct 2023 23:59:59 +0000
Start : Tue, 14 Nov 2023 00:00:00 +0000
End : Mon, 20 Nov 2023 23:59:59 +0000
```

Reply:
> latest changes now match those values. I believe the UI still matches though the dates are in local TZ (EST) so the end day is a day too early on the UI in EST time zone.

This is going to be a problem though, someone in Europe hitting CR is going to get different dates than what metrics is caching, and they're going to see inconsistent results from someone in the US. 

Can we modify the code to only extract the month, day, and year, and then construct a UTC date from these elements? For instance, if a user selects October 31, 2023, it's currently converted from their local time to UTC today. Instead, could we directly use the values '10', '31', and '23' to create a UTC date? I find handling dates and times in JavaScript challenging, but I think we need to try to make sure it's the same for everyone regardless of where they are

Reply:
> wait, that first one isn't matching

Yes, you're still at 29 something days, chatgpt is good at telling you how long a date range is (it'll just make a python script like below) 

```
from datetime import datetime

# Time stamps
start_time = datetime.strptime("Mon, 02 Oct 2023 00:00:00 +0000", "%a, %d %b %Y %H:%M:%S %z")
end_time = datetime.strptime("Mon, 30 Oct 2023 23:59:59 +0000", "%a, %d %b %Y %H:%M:%S %z")

# Calculate difference
time_difference = end_time - start_time
time_difference.days, time_difference.seconds
```

Reply:
I put some logic/logging in the metrics.go file so that we can log the number of days for easier debugging.  This looks like what we want:

```
time="2023-11-21T14:04:09.467-05:00" level=debug msg="most recent GA is \"4.14\""
time="2023-11-21T14:04:09.467-05:00" level=info msg="Start : Wed, 04 Oct 2023 00:00:00 +0000"
time="2023-11-21T14:04:09.467-05:00" level=info msg="End   : Tue, 31 Oct 2023 23:59:59 +0000"
time="2023-11-21T14:04:09.467-05:00" level=info msg="diff  : 28.00 days"
time="2023-11-21T14:04:09.467-05:00" level=info msg="int   : 2419199 seconds"
time="2023-11-21T14:04:11.883-05:00" level=info msg="Start : Wed, 15 Nov 2023 00:00:00 +0000"
time="2023-11-21T14:04:11.883-05:00" level=info msg="End : Tue, 21 Nov 2023 23:59:59 +0000"
time="2023-11-21T14:04:11.883-05:00" level=info msg="diff  : 7.00 days"
time="2023-11-21T14:04:11.883-05:00" level=info msg="int   : 604799 seconds"
```

I pulled the year/month/day out of the date (returned by the DatePicker) as a string and passed that along instead of what the DatePicker returns.

I also removed the `Z` from the ga_dates as this was causing the `new Date()` invocation to convert the timezone to my local timezone.  If you load up CR, you can now see the GA date for 4.14 as 10/31/2023 instead of 10/30/2023

Reply:
Metrics does give me the values I expect:

```
DEBU[2023-11-21T14:20:47.35-05:00] most recent GA is "4.14"
INFO[2023-11-21T14:20:47.35-05:00] Start : Wed, 04 Oct 2023 00:00:00 +0000
INFO[2023-11-21T14:20:47.35-05:00] End   : Tue, 31 Oct 2023 23:59:59 +0000
INFO[2023-11-21T14:20:47.35-05:00] diff  : 28.00 days
INFO[2023-11-21T14:20:47.35-05:00] int   : 2419199 seconds
INFO[2023-11-21T14:20:47.35-05:00] Start : Wed, 15 Nov 2023 00:00:00 +0000
INFO[2023-11-21T14:20:47.35-05:00] End : Tue, 21 Nov 2023 23:59:59 +0000
INFO[2023-11-21T14:20:47.35-05:00] diff  : 7.00 days
INFO[2023-11-21T14:20:47.35-05:00] int   : 604799 seconds
```

But the UI is still off by one:

```
baseStartTime = '2023-10-03 00:00:00'
baseEndTime = '2023-10-31 23:59:59'


sampleStartTime = '2023-11-14 00:00:00'
sampleEndTime = '2023-11-21 23:59:59'
 ​
```